### PR TITLE
Fix a bunch of tests, cleanup :fish: and more

### PR DESCRIPTION
**Pull request by knyazer**

### CODE REVIEW

**Review by ry2009 — COMMENTED** [review]
Took a look. SRO attention logic is clean, tiled associative scan with online softmax combine is the right pattern.

Two small things:

1. `XLA_FLAGS = "--xla_gpu_deterministic_ops=true"` in pyproject.toml is test-only right? Just want to confirm this doesn't leak into prod inference configs since it has perf implications.

2. `StableReductionOrderAttention` is the default in `AttentionForwardPassConfig` -- means all callers get the tiled path unless they explicitly pass `DefaultAttention()`... Intentional for correctness, just flagging since it's a behavior change for existing users who don't pass a config.

Otherwise LGTM. The hacky forward_pass_config threading is fine given the refactor PR will clean it up:) well done imo

File: lalamo/modules/token_mixers/attention.py [hunk 1]
```diff
@@ -81,6 +85,167 @@ def _soft_capped_attention_kernel(
...
+        combine,
+        (tile_max, tile_sum, tile_output),
+    )
+
+    result = final_output[-1] / final_sum[-1, ..., None]
+    return rearrange(result, "heads queries hidden -> queries heads hidden").astype(original_dtype)
+
+
+@dataclass(frozen=True)
+class DefaultAttention:
```
Comment (norpadon) [inline lalamo/modules/token_mixers/attention.py]: It should have a name that makes it clear that it is a part fof the config and not a layer

File: lalamo/modules/token_mixers/attention.py [hunk 2]
```diff
@@ -425,24 +590,39 @@ def __call__(
...
+                else:
+                    attention_output = jax.nn.dot_product_attention(
+                        queries,
+                        updated_state.keys,
+                        updated_state.values,
+                        bias=sink_bias,
+                        mask=mask,
+                        scale=self.scale,
+                    )
+            case StableReductionOrderAttention(tile_size=tile_size, upcast_dtype=upcast_dtype):
```
Comment (norpadon) [inline lalamo/modules/token_mixers/attention.py]: Maybe it is a good idea to make the kernel itself a method of the config, so that we don't need this match statement

**Review by norpadon — CHANGES_REQUESTED** [review]

File: lalamo/modules/token_mixers/attention.py (hunk 1 above)
Comment (norpadon) [reply lalamo/modules/token_mixers/attention.py]: DefaultAttentionMode or something

File: lalamo/modules/token_mixers/attention.py [hunk 3]
```diff
@@ -363,7 +528,7 @@ def __call__(
         state: KVCacheLayer | None = None,
         return_updated_state: bool = False,
         length_without_padding: Int[Array, ""] | int | None = None,
-        forward_pass_config: MixerForwardPassConfig = MixerForwardPassConfig(),  # noqa: ARG002, B008
+        forward_pass_config: AttentionForwardPassConfig | None = None,
```
Comment (norpadon) [inline lalamo/modules/token_mixers/attention.py:562]: Oh no, this is very wrong. MixerForwardPassConfig should include attention mode inside of it. Attention should still receive MixerForwardPassConfig. Forward pass configs have static shape

**Review by norpadon — CHANGES_REQUESTED** [review]

**Comment by knyazer** [comment]
Closing as mostly integrated by parts into upstream or refactor already
